google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableInstanceAdminClientIT.java

```diff
@@ -0,0 +1,192 @@
+import com.google.cloud.bigtable.test_helpers.env.TestEnvRule;
+import java.io.IOException;
+import java.util.List;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class BigtableInstanceAdminClientIT {
+
+  @ClassRule public static TestEnvRule testEnvRule = new TestEnvRule();
+
+  private String projectId = testEnvRule.env().getProjectId();
+  private String instanceId = testEnvRule.env().getInstanceId();
+  private BigtableInstanceAdminClient client;
+
+  @BeforeClass
+  public static void validatePlatform() {
+    assume()
+        .withMessage("BigtableInstanceAdminClient doesn't support on Emulator")
+        .that(testEnvRule.env())
+        .isNotInstanceOf(EmulatorEnv.class);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    client = BigtableInstanceAdminClient.create(projectId);
+  }
+
+  @Test
+  public void appProfileTest() {
+    String testAppProfile = "testAP-" + RandomStringUtils.randomAlphanumeric(5);
+
+    Cluster cluster = client.listClusters(instanceId).get(0);
+
+    AppProfile newlyCreatedAppProfile =
+        client.createAppProfile(
+            CreateAppProfileRequest.of(instanceId, testAppProfile)
+                .setRoutingPolicy(AppProfile.SingleClusterRoutingPolicy.of(cluster.getId()))
+                .setDescription("This is a test app profile"));
+
+    AppProfile updated =
+        client.updateAppProfile(
+            UpdateAppProfileRequest.of(newlyCreatedAppProfile)
+                .setDescription("This is new app profile description"));
+
+    AppProfile freshAppProfile = client.getAppProfile(instanceId, testAppProfile);
+    assertThat(freshAppProfile.getDescription()).isEqualTo(updated.getDescription());
+
+    assertThat(client.listAppProfiles(instanceId)).contains(freshAppProfile);
+
+    // Deleting appProfile fails with below message
+    // io.grpc.StatusRuntimeException: FAILED_PRECONDITION: Deleting an app profile will cause all
+    // requests using it to fail.
+    //    client.deleteAppProfile(instanceId, testAppProfile);
+  }
+
+  @Test
+  public void iamUpdateTest() {
+    // TODO(rahulkql): confirm this test as it involes assigning roles.
+    //    Policy newPolicy = client.setIamPolicy(
+    //        instanceId,
+    //        Policy.newBuilder()
+    //            .setEtag("")
+    //            .addIdentity(Role.viewer(), Identity.user("test@example.com"))
+    //            .build());
+
+    assertThat(client.getIamPolicy(instanceId)).isNotNull();
+
+    List<String> permissions =
+        client.testIamPermission(
+            instanceId, "bigtable.tables.readRows", "bigtable.tables.mutateRows");
+    assertThat(permissions).hasSize(2);
+  }
+
+  // TODO(rahulkql): Yet to run this test.
+  /* Please set createResource=true to run this. */
+  @Test
+  public void instanceCreationDeletionTest() {
```

Comment:
@igorbernstein2 I would like to know your opinion before running this as it would be an expensive test.

Reply:
Thanks for the response, I have remove the need for another property.